File: 169-MajorityElement/169-MajorityElement.java
// Last updated: 31/07/2025, 21:18:08
class Solution {
    public int majorityElement(int[] nums) {
        // Arrays.sort(nums);
        // return nums[nums.length/2];
       return  more_than_half(nums);
    }
    public static int more_than_half(int[] arr)
    {
        int len=arr.length;
        int v=1;
        int e=arr[0];
        for(int i=1;i<len;i++)
        {
            if(e==arr[i])
            {
                v+=1;
            }
            else
            {
                v--;
                if(v==0)
                {
                    e=arr[i];
                    v=1;
                }
            }    
        }
        return e;
    }
}

File: 169-MajorityElement/169-MajorityElement.py
# Last updated: 31/07/2025, 21:18:14
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n=len(nums)/2
        dic={}
        for i in nums:
            if i not in dic:
                dic[i]=nums.count(i)
                if dic[i]>n:
                    return i


        